apps/src/lib/levelbuilder/lesson-editor/AddLevelDialog.jsx

```diff
@@ -60,10 +62,58 @@ export default class AddLevelDialog extends Component {
     super(props);
 
     this.state = {
-      methodOfAddingLevel: 'Find Level'
+      methodOfAddingLevel: 'Find Level',
+      levels: null,
+      searchFields: null,
+      currentPage: 1
     };
   }
 
+  componentDidMount() {
+    $.ajax({
+      url: `/levels/get_filters`,
+      method: 'GET',
+      contentType: 'application/json;charset=UTF-8'
+    }).done((data, _, request) => {
+      this.setState({searchFields: data});
+    });
+
+    $.ajax({
+      url: `/levels/get_filtered_levels?page=${this.state.currentPage}`,
+      method: 'GET',
+      contentType: 'application/json;charset=UTF-8'
+    }).done((data, _, request) => {
+      this.setState({levels: data});
```

Comment:
since you're not using `request` either, this can just be `data => ...`. If you prefer to list all params, it seems like you should prefix both of them with `_`, like `(data, _foo, _request) => ...`.